lib/reverse_sentence.rb

```diff
@@ -1,6 +1,37 @@
 # A method to reverse the words in a sentence, in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: linear, O(n) - where n is the length of input string
+# Space complexity: constant, O(1) - the space used remains the same regardless
+#                   of the input size
 def reverse_sentence(my_sentence)
-  raise NotImplementedError
+  return nil if my_sentence == nil
+
+  start_index = 0
+  end_index = 0
+  length = my_sentence.length
+
+  while end_index < length
+    if my_sentence[end_index] == " " || end_index == length - 1
```

Comment:
Tiny nit - this conditional can be a bit confusing at first glance. Maybe put it into a separate variable, e.g. `is_word_ending` (and then using that variable in the `if` statement).